### ISSUE TITLE
Cannot reproduce COIN dataset result

### ISSUE THREAD

**Issue by bluehawk2k** [showlab/videollm-online#26]
Hi, here is another issue about reproducibility of COIN dataset result.
I've also tried to reproduce your result of COIN dataset with using 8 A100 GPUs.
However, the evaluation result gives too low performance comparing with your result reported in your paper.
![image](https://github.com/user-attachments/assets/3d165c7e-cf4c-4b4b-b4df-6dfe15f5723c)
Do you have any idea about this issue?

**Comment by leebebeto** [comment]
Hi, I also found a part that may have potential bugs in the evaluation part when using Coin dataset.
In evaluation_loop() of ./transformers/trainer.py, the labels seem to be wrong.
Following is the screenshot of the logits and labels. 
I think the labels should also look like logits, indicating the gt_ids, not 0.
Because of the wrong labels, I also get very low accuracy. 

<img width="829" alt="image" src="https://github.com/user-attachments/assets/0280601d-fe7c-4a0d-ad5d-b72f6d0b6ea0">

Thanks in advance!

**Comment by chenjoya** [comment]
Hi @bluehawk2k , it seems that the loss is too large and the model is not converged. Could you show your training scripts? Sometimes on different devices, the training in the same learning rate may not be stable again, you can decrease the learning rate or extend the training epochs a little bit. When my model is converged, the loss is in 0.0x scale.

@leebebeto the labels here is not the real label, it is used as the dataset index.

**Comment by chenjoya** [comment]
Recently I am busy with other projects. But please feel free to leave your questions here and I will solve them as soon as possible.

**Comment by leebebeto** [comment]
Thank you for the reply.
By the way, I also have loss value converging to 0.x scale rather than 0.0x scale.
I used your released training script https://github.com/showlab/videollm-online/blob/main/scripts/coin/live1%2B.sh
Do I have to decrease learning rates or extend training epochs from your released training script?

Thank you!

**Comment by chenjoya** [comment]
Hi, currently I am working in a new cluster. I finished all data preparing and now start to train COIN. Please wait several hours and I will get back to you!

<img width="909" alt="image" src="https://github.com/user-attachments/assets/315319d2-0fcd-4966-927e-07d5ea1736a2">

**Comment by chenjoya** [comment]
Hi, the bug in COIN evaluation has been fixed. I reproduced from scratch in a new cluster. 

The main reason should be the unstable training. My current env is torch 2.4.0, cuda 12.4, transformers 4.44.0. When I try to use lr = 2e-4, it cannot converge, which I also meet in other scenarios. I try to decrease the lr / 2 = 1e-4, still 5 epoch, then get the results

<img width="1297" alt="image" src="https://github.com/user-attachments/assets/02a9c719-e455-4f0b-b8e5-8a867df5f67e">

Lower than the paper due to lr / 2. Will update once I get a good epoch and lr parameters. 

Meanwhile, I made some updates on the COIN dataset and removed some useless codes. Recommend to pull them.

**Comment by chenjoya** [comment]
Hi, sorry that I recently busy with other projects, do not have many GPUs to re-implement. In my experience, the key to get high accuracy on COIN is (1) avoid training loss spike. This is very important, please check the log. If that happens, you can stop and try the next config 😂 (2) Try lr as high as possible but ensure loss spike is not appeared.

Yesterday I tried some simple parameters, lr = 0.00015, stream_loss_weight=0.5, and I have got the similar results to the paper: 

<img width="1248" alt="image" src="https://github.com/user-attachments/assets/e0688a77-79b7-46e5-8206-3112985747bd">

‘eval_coin_step_test_accuracy’: 62.7394328517924
’eval_coin_next_test_accuracy’: 48.5977275995973
‘eval_coin_task_test_accuracy’: 92.22408026755853
‘eval_coin_procedure_test_accuracy’: 48.84329225761451
‘eval_coin_taskprocedure_test_accuracy’: 53.641383318802674

Training with longer epochs should the match the paper. But now my GPUs are not free again. Will update once it has been done. 

Many thanks for your patience!

**Comment by yankee624** [comment]
@chenjoya Do you think the training script for ego4d narration (https://github.com/showlab/videollm-online/blob/main/scripts/ego4d/narration/live1%2B.sh) should also be fixed? I'm trying to train the model on `ego4d_refined_narration_stream_train`, but the loss isn't going down after it reaches 0.3~0.4 in the middle of epoch 1.
![image](https://github.com/user-attachments/assets/4fbeea28-96ba-4b50-87d7-433ac964c300)

**Comment by chenjoya** [comment]
hello @yankee624 this is very good since there is no loss spike. The loss in training COIN is low because it just contains single video-text pairs, instead of multiple video-text streams in Ego4D narration.

**Comment by yankee624** [comment]
@chenjoya Thank you so much for confirming! The metrics seem good so I guess it's trained well!

**Comment by nguyentthong** [comment]
hi @chenjoya , can I ask how many test samples you evaluate on? Because when I evaluated on 512 samples, the test step accuracy is 65%. However, the test step accuracy on 10k samples is only 10%.

**Comment by chenjoya** [comment]
hi @nguyentthong , I tested on the standard test set of COIN dataset. Do you get the correct video ids?

**Comment by nguyentthong** [comment]
hi @chenjoya , I use the official data in this link: https://github.com/coin-dataset/annotations/blob/master/COIN.json

My loss landscape is as follows:

![image](https://github.com/user-attachments/assets/edd7fbef-6628-4279-bdad-dfbd503595dd)

Is this a normal loss landscape?

**Comment by nguyentthong** [comment]
Additionally, I use the preprocess scripts here: https://github.com/showlab/videollm-online/tree/main/data/preprocess

I suppose this is applicable for both COIN and Ego4D datasets. Can you tell me whether my assumption is correct? @chenjoya

**Comment by SeongRyong0726** [comment]
I also use the coin.json file from  https://github.com/coin-dataset/annotations/blob/master/COIN.json .
However, I observe that in the COIN class (/data/coin/coin.py) it detect video_uid as video_uid (but in the COIN.json, it does not exist), and 'train' and 'test' as 'subset' (but in the file, those are "training" and "testing").
I thought that is official file (uploaded years ago), or is there another file for this repo?

Thank you in advance..!
